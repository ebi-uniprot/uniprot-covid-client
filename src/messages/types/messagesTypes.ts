import { Location } from '../../urls';

export enum MessageLevel {
  SUCCESS = 'success',
  INFO = 'info',
  WARNING = 'warning',
  FAILURE = 'failure',
}

export enum MessageFormat {
  POP_UP = 'POP_UP', // eg pops up at the bottom right of the page
  IN_PAGE = 'IN_PAGE', // eg full width banner at the top of the page
}

export enum MessageTag {
  BLAST_JOB = 'BLAST_JOB',
  DOWNTIME = 'DOWNTIME',
}

export type MessageType = {
  id: string;
  content: string | JSX.Element;
  format: MessageFormat;
  level: MessageLevel;
  dateActive?: number;
  dateExpired?: number;
  tag?: MessageTag;
  locations?: Location[];
};
