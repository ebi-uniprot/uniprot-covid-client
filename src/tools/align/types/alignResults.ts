/* Results as given by the server */
/* and also adding the results afters parsing because server just gives text */

export type AlignResults = string;

export type PhyloTreeNode = {
  name?: string;
  distance?: number;
  distanceFromRoot?: number;
  children?: PhyloTreeNode[];
};
export type PhyloTree = PhyloTreeNode;

export type PIMRow = { name: string; accession?: string; values: number[] };
export type PIM = PIMRow[];

export type AlnClustalNum = {
  sequences: {
    sequence: string;
    name: string;
  }[];
  conservation: string;
};
