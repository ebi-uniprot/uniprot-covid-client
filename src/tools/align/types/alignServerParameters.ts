/* Parameters of a blast job as required by the server */
// https://wwwdev.ebi.ac.uk/Tools/services/rest/clustalo?wadl
// https://wwwdev.ebi.ac.uk/Tools/services/rest/clustalo/parameters
// parameter-specific documentation at
// https://wwwdev.ebi.ac.uk/Tools/services/rest/clustalo/parameterdetails/<parameter_name>

export type ServerParameters = {
  email: string; // email
  // A title to identify the analysis job (managed by the server, different
  // than our job name)
  title?: string;
  guidetreeout?: boolean;
  dismatout?: boolean;
  dealign?: boolean;
  mbed?: boolean;
  mbediteration?: boolean;
  iterations?: 0 | 1 | 2 | 3 | 4 | 5;
  gtiterations?: number;
  hmmiterations?: number;
  outfmt?: string;
  order?: 'aligned' | 'input';
  stype?: string;
  sequence: string; // sequence
};

// same, but without the email, as the server will not send that with the rest
// of the input params
export type PublicServerParameters = Omit<ServerParameters, 'email'>;
