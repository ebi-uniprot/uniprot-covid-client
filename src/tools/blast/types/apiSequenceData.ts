import { EntryType } from '../../../uniprotkb/adapters/uniProtkbConverter';

type Name = {
  value: string;
};

type ProteinName = {
  fullName: Name;
  shortNames?: Name[];
};

export type APISequenceData = {
  entryType: EntryType;
  primaryAccession: string;
  uniProtkbId: string;
  entryAudit: {
    sequenceVersion: number;
  };
  organism: {
    scientificName: string;
    commonName: string;
    taxonId: number;
    lineage: string[];
  };
  proteinExistence: string;
  proteinDescription: {
    recommendedName: ProteinName;
    alternativeNames: ProteinName[];
  };
  genes: {
    geneName: Name;
    synonyms: Name[];
  }[];
  sequence: {
    value: string;
    length: number;
    molWeight: number;
    crc64: string;
    md5: string;
  };
};
