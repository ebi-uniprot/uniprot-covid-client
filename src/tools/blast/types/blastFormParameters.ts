import {
  SType,
  Database,
  Exp,
  Matrix,
  Filter,
  GapAlign,
  Scores,
  Alignments,
  Program,
  Sequence,
} from './blastServerParameters';
import { SelectedTaxon } from '../config/BlastFormData';

export type FormParameters = {
  stype: SType;
  program: Program;
  sequence: Sequence;
  database: Database;
  taxIDs: SelectedTaxon[];
  threshold: Exp;
  matrix: Matrix;
  filter: Filter;
  gapped: GapAlign;
  hits: Scores | Alignments;
};
