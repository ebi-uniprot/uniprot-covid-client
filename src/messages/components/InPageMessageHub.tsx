import React, { FC } from 'react';
import { Message } from 'franklin-sites';
import { MessageType } from '../types/messagesTypes';

type InPageMessageHubProps = {
  messages: MessageType[];
  onDismiss: (id: string) => void;
};

const InPageMessageHub: FC<InPageMessageHubProps> = ({
  messages = [],
  onDismiss,
}) => (
  <div>
    {messages.map(({ level, content, id }) => (
      <Message key={id} level={level} onDismiss={() => onDismiss(id)}>
        {content}
      </Message>
    ))}
  </div>
);

export default InPageMessageHub;
