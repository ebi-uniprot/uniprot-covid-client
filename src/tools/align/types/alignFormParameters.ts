import { ServerParameters } from './alignServerParameters';

export type FormParameters = {
  sequence: ServerParameters['sequence'];
  order: ServerParameters['order'];
  iterations: ServerParameters['iterations'];
};
