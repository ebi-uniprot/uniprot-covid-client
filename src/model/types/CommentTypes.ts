import { Evidence } from './modelTypes';
import {
  Absorption,
  KineticParameters,
  CofactorComment,
} from '../uniprotkb/sections/FunctionConverter';
import { FeatureData } from '../../view/uniprotkb/components/FeaturesView';
import { Interactant } from '../uniprotkb/sections/InteractionConverter';

export enum CommentType {
  ACTIVITY_REGULATION = 'ACTIVITY REGULATION',
  ALLERGEN = 'ALLERGEN',
  ALTERNATIVE_PRODUCTS = 'ALTERNATIVE PRODUCTS',
  BIOPHYSICOCHEMICAL_PROPERTIES = 'BIOPHYSICOCHEMICAL PROPERTIES',
  BIOTECHNOLOGY = 'BIOTECHNOLOGY',
  CATALYTIC_ACTIVITY = 'CATALYTIC ACTIVITY',
  CAUTION = 'CAUTION',
  COFACTOR = 'COFACTOR',
  DEVELOPMENTAL_STAGE = 'DEVELOPMENTAL STAGE',
  DISEASE = 'DISEASE',
  DISRUPTION_PHENOTYPE = 'DISRUPTION PHENOTYPE',
  DOMAIN = 'DOMAIN',
  FUNCTION = 'FUNCTION',
  INDUCTION = 'INDUCTION',
  INTERACTION = 'INTERACTION',
  MASS_SPECTROMETRY = 'MASS SPECTROMETRY',
  MISCELLANEOUS = 'MISCELLANEOUS',
  PATHWAY = 'PATHWAY',
  PHARMACEUTICAL = 'PHARMACEUTICAL',
  POLYMORPHISM = 'POLYMORPHISM',
  PTM = 'PTM',
  RNA_EDITING = 'RNA EDITING',
  SEQUENCE_CAUTION = 'SEQUENCE CAUTION',
  SIMILARITY = 'SIMILARITY',
  SUBCELLULAR_LOCATION = 'SUBCELLULAR LOCATION',
  SUBUNIT = 'SUBUNIT',
  TISSUE_SPECIFICITY = 'TISSUE SPECIFICITY',
  TOXIC_DOSE = 'TOXIC DOSE',
  UNKNOWN = 'UNKNOWN',
  WEB_RESOURCE = 'WEB RESOURCE',
}

export type FreeTextType =
  | CommentType.DISRUPTION_PHENOTYPE
  | CommentType.DOMAIN
  | CommentType.FUNCTION
  | CommentType.INDUCTION
  | CommentType.MISCELLANEOUS
  | CommentType.PATHWAY
  | CommentType.PTM
  | CommentType.SIMILARITY
  | CommentType.SUBUNIT
  | CommentType.TISSUE_SPECIFICITY
  | CommentType.POLYMORPHISM;

export type TextWithEvidence = {
  value: string;
  evidences?: Evidence[];
  id: string;
};

export type FreeTextComment = {
  commentType: FreeTextType;
  texts?: TextWithEvidence[];
  molecule?: string;
};

export type AbsorptionComment = {
  commentType: CommentType.BIOPHYSICOCHEMICAL_PROPERTIES;
  absorption?: Absorption;
};

export type KineticsComment = {
  commentType: CommentType.BIOPHYSICOCHEMICAL_PROPERTIES;
  kineticParameters?: KineticParameters;
};

export type pHDependenceComment = {
  commentType: CommentType.BIOPHYSICOCHEMICAL_PROPERTIES;
  phDependence: {
    texts: TextWithEvidence[];
  };
};

export type RedoxPotentialComment = {
  commentType: CommentType.BIOPHYSICOCHEMICAL_PROPERTIES;
  redoxPotential: {
    texts: TextWithEvidence[];
  };
};

export type TemperatureDependenceComment = {
  commentType: CommentType.BIOPHYSICOCHEMICAL_PROPERTIES;
  temperatureDependence: {
    texts: TextWithEvidence[];
  };
};

export type CatalyticActivityComment = {
  commentType: CommentType.CATALYTIC_ACTIVITY;
  reaction?: {
    name: string;
    reactionCrossReferences?: { database: string; id: string }[];
    ecNumber: string;
    evidences?: Evidence[];
  };
  physiologicalReactions?: PhysiologicalReaction[];
};

export enum PhysiologicalReactionDirection {
  LeftToRight = 'left-to-right',
  RightToLeft = 'right-to-left',
}

export type PhysiologicalReaction = {
  directionType: PhysiologicalReactionDirection;
  reactionCrossReference: { database: string; id: string };
  evidences: Evidence[];
};

export type Xref = {
  database?: string;
  id?: string;
  properties?: { [key: string]: string };
  additionalIds?: string[];
  isoformId?: string;
  implicit?: true;
};

export type DiseaseType = {
  diseaseId?: string;
  diseaseAccession?: string;
  acronym?: string;
  description?: string;
  diseaseCrossReference?: Xref;
  evidences?: Evidence[];
};

export type DiseaseComment = {
  commentType: CommentType.DISEASE;
  disease?: DiseaseType;
  note?: { texts?: { value?: string }[] };
};

export enum InteractionType {
  SELF = 'SELF',
  XENO = 'XENO',
  BINARY = 'BINARY',
  // Are there others?
}

export type Interaction = {
  numberOfExperiments: number;
  type: InteractionType;
  interactantOne: Interactant;
  interactantTwo: Interactant;
};

export type InteractionComment = {
  commentType: CommentType.INTERACTION;
  interactions: Interaction[];
};

export type Isoform = {
  name: { value: string };
  isoformSequenceStatus: string;
  isoformIds: string[];
  synonyms: { value: string }[];
  note?: { texts: TextWithEvidence[] };
  sequenceIds: string[];
  varSeqs: FeatureData;
};

export type AlternativeProductsComment = {
  commentType: CommentType.ALTERNATIVE_PRODUCTS;
  isoforms: Isoform[];
  note?: { texts: TextWithEvidence[] };
  events: string[];
};

export type SequenceCautionComment = {
  commentType: CommentType.SEQUENCE_CAUTION;
  sequenceCautionType: string;
  sequence: string;
  note?: string;
  evidences?: Evidence[];
};

export type MassSpectrometryComment = {
  commentType: CommentType.MASS_SPECTROMETRY;
  molecule?: string;
  method?: string;
  note?: string;
  molWeight: number;
  molWeightError: number;
  evidences: Evidence[];
};

export type RNAEditingComment = {
  commentType: CommentType.RNA_EDITING;
  locationType?: string;
  positions: { position: number; evidences: Evidence[] }[];
  note?: { texts: TextWithEvidence[] };
};

export type SubcellularLocationComment = {
  commentType: CommentType.SUBCELLULAR_LOCATION;
  molecule?: string;
  note?: { texts: TextWithEvidence[] };
  subcellularLocations?: {
    location: TextWithEvidence;
    topology?: TextWithEvidence;
    orientation?: TextWithEvidence;
  }[];
};

export type WebResourceComment = {
  commentType: CommentType.WEB_RESOURCE;
  note?: string;
  resourceName: string;
  resourceUrl: string;
};

type Comment =
  | FreeTextComment
  | CatalyticActivityComment
  | DiseaseComment
  | InteractionComment
  | AlternativeProductsComment
  | SequenceCautionComment
  | SubcellularLocationComment
  | MassSpectrometryComment
  | RNAEditingComment
  | AbsorptionComment
  | KineticsComment
  | CofactorComment
  | pHDependenceComment
  | RedoxPotentialComment
  | TemperatureDependenceComment
  | SubcellularLocationComment
  | WebResourceComment;

export default Comment;
