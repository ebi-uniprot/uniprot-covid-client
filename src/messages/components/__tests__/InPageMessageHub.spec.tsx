import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import InPageMessageHub from '../InPageMessageHub';
import { MessageFormat, MessageLevel } from '../../types/messagesTypes';

const messages = [
  {
    id: 'msg1',
    content: 'in page message',
    format: MessageFormat.IN_PAGE,
    level: MessageLevel.INFO,
  },
];

describe('InPageMessageHub tests', () => {
  test('it should render and call dismiss', () => {
    const onDismissFn = jest.fn();
    const { asFragment, getByRole } = render(
      <InPageMessageHub messages={messages} onDismiss={onDismissFn} />
    );
    const button = getByRole('button');
    fireEvent.click(button);
    expect(onDismissFn).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
