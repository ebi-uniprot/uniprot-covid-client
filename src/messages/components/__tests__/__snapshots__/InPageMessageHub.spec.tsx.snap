// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InPageMessageHub tests it should render and call dismiss 1`] = `
<DocumentFragment>
  <div>
    <div
      class="message message--info"
      role="status"
    >
      <section
        class="message__content"
      >
        <svg
          fill="none"
          height="1.125em"
          viewBox="0 0 18 18"
          width="1.125em"
        >
          <path
            clip-rule="evenodd"
            d="M18 9A9 9 0 110 9a9 9 0 0118 0zM8 4a1 1 0 112 0v6a1 1 0 11-2 0V4zm1 9a1 1 0 100 2 1 1 0 000-2z"
            fill="#000"
            fill-rule="evenodd"
          />
        </svg>
        <small>
          in page message
        </small>
      </section>
      <button
        class="message__dismiss"
        type="button"
      >
        <svg
          height="10"
          viewBox="0 0 352 512"
          width="10"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
